==426== Helgrind, a thread error detector
==426== Copyright (C) 2007-2017, and GNU GPL'd, by OpenWorks LLP et al.
==426== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==426== Command: ./philo 2 1000 100 100 3
==426== 
==426== ---Thread-Announcement------------------------------------------
==426== 
==426== Thread #3 was created
==426==    at 0x499C122: clone (clone.S:71)
==426==    by 0x48612EB: create_thread (createthread.c:101)
==426==    by 0x4862E0F: pthread_create@@GLIBC_2.2.5 (pthread_create.c:817)
==426==    by 0x4842917: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x1099F2: make_threads (init.c:63)
==426==    by 0x109478: main (main.c:48)
==426== 
==426== ---Thread-Announcement------------------------------------------
==426== 
==426== Thread #2 was created
==426==    at 0x499C122: clone (clone.S:71)
==426==    by 0x48612EB: create_thread (createthread.c:101)
==426==    by 0x4862E0F: pthread_create@@GLIBC_2.2.5 (pthread_create.c:817)
==426==    by 0x4842917: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x1099F2: make_threads (init.c:63)
==426==    by 0x109478: main (main.c:48)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x4864FF9: pthread_mutex_lock (pthread_mutex_lock.c:81)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x486500C: pthread_mutex_lock (pthread_mutex_lock.c:159)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x4864FF9: pthread_mutex_lock (pthread_mutex_lock.c:81)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x486500C: pthread_mutex_lock (pthread_mutex_lock.c:159)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A024: check_death (routine.c:111)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A024: check_death (routine.c:111)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A024: check_death (routine.c:111)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A499: death_log (log_msg.c:43)
==426==    by 0x10A04A: check_death (routine.c:112)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A499: death_log (log_msg.c:43)
==426==    by 0x10A04A: check_death (routine.c:112)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A499: death_log (log_msg.c:43)
==426==    by 0x10A04A: check_death (routine.c:112)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x4864FF9: pthread_mutex_lock (pthread_mutex_lock.c:81)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A3DF: task_log (log_msg.c:33)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x486500C: pthread_mutex_lock (pthread_mutex_lock.c:159)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A3DF: task_log (log_msg.c:33)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A3DF: task_log (log_msg.c:33)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A3DF: task_log (log_msg.c:33)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A437: task_log (log_msg.c:36)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A437: task_log (log_msg.c:36)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A437: task_log (log_msg.c:36)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== Use --history-level=approx or =none to gain increased speed, at
==426== the cost of reduced accuracy of conflicting-access information
==426== ERROR SUMMARY: 13196 errors from 33 contexts (suppressed: 0 from 0)
==426== 
==426== 1 errors in context 1 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A437: task_log (log_msg.c:36)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 2 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A437: task_log (log_msg.c:36)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 3 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A437: task_log (log_msg.c:36)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 4 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A3DF: task_log (log_msg.c:33)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 5 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A3DF: task_log (log_msg.c:33)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 6 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x486500C: pthread_mutex_lock (pthread_mutex_lock.c:159)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A3DF: task_log (log_msg.c:33)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 7 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x4864FF9: pthread_mutex_lock (pthread_mutex_lock.c:81)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A3DF: task_log (log_msg.c:33)
==426==    by 0x10A0DE: philo_sleep (routine.c:129)
==426==    by 0x10A14C: routine (routine.c:146)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 8 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 9 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 10 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 11 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A499: death_log (log_msg.c:43)
==426==    by 0x10A04A: check_death (routine.c:112)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 12 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A499: death_log (log_msg.c:43)
==426==    by 0x10A04A: check_death (routine.c:112)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 13 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A499: death_log (log_msg.c:43)
==426==    by 0x10A04A: check_death (routine.c:112)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 14 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A024: check_death (routine.c:111)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 15 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A024: check_death (routine.c:111)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 1 errors in context 16 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A024: check_death (routine.c:111)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 2 errors in context 17 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 2 errors in context 18 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 2 errors in context 19 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FD3: check_death (routine.c:105)
==426==    by 0x109EF4: philo_eat (routine.c:83)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 923 errors in context 20 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 923 errors in context 21 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 923 errors in context 22 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x486500C: pthread_mutex_lock (pthread_mutex_lock.c:159)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 923 errors in context 23 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x4864FF9: pthread_mutex_lock (pthread_mutex_lock.c:81)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 928 errors in context 24 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 928 errors in context 25 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 928 errors in context 26 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 929 errors in context 27 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 929 errors in context 28 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #3
==426== Locks held: none
==426==    at 0x486500F: pthread_mutex_lock (pthread_mutex_lock.c:161)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 929 errors in context 29 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x486500C: pthread_mutex_lock (pthread_mutex_lock.c:159)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 929 errors in context 30 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE0 by thread #3
==426== Locks held: none
==426==    at 0x4864FF9: pthread_mutex_lock (pthread_mutex_lock.c:81)
==426==    by 0x483FE78: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x109FAC: check_death (routine.c:102)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 994 errors in context 31 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE0 by thread #2
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667D1: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:52)
==426==    by 0x48667D1: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee0 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 994 errors in context 32 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during write of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== 
==426== 994 errors in context 33 of 33:
==426== ----------------------------------------------------------------
==426== 
==426== Possible data race during read of size 4 at 0x1FFEFFFEE4 by thread #2
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x10A2D9: my_sleep (utils.c:53)
==426==    by 0x109F4C: philo_eat (routine.c:88)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==    by 0x499C132: clone (clone.S:95)
==426== 
==426== This conflicts with a previous write of size 4 by thread #3
==426== Locks held: none
==426==    at 0x48667CB: __pthread_mutex_unlock_usercnt (pthread_mutex_unlock.c:55)
==426==    by 0x48667CB: pthread_mutex_unlock (pthread_mutex_unlock.c:357)
==426==    by 0x4840458: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x10A065: check_death (routine.c:115)
==426==    by 0x109E56: pick_forks (routine.c:50)
==426==    by 0x109EE8: philo_eat (routine.c:82)
==426==    by 0x10A132: routine (routine.c:142)
==426==    by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
==426==    by 0x4862608: start_thread (pthread_create.c:477)
==426==  Address 0x1ffefffee4 is on thread #1's stack
==426==  in frame #3, created by main (main.c:33)
==426== 
==426== ERROR SUMMARY: 13196 errors from 33 contexts (suppressed: 0 from 0)
